# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: layout/style/nsStyleTransformMatrix.cpp
# Commit: ec44194caefb
# Full Hash: ec44194caefb41931247d18ad886326cf6d89a93
# Author: Boris Chiou <boris.chiou@gmail.com>
# Date: 2024-03-05 09:48:50
# Description:
#   Bug 1870200 - Break the cyclic dependency if any of the desendants uses non-scaling-stroke. r=emilio
#   
#   While we are computing the transform-box:stroke-box, for CSS Transforms or
#   Motion path Transforms, we have to avoid the cyclic dependency not only
#   for the SVG geometry frame itself, but also for the desendants of
# ==============================================================================

diff -r 5c84d062c789 -r ec44194caefb layout/style/nsStyleTransformMatrix.cpp
--- a/layout/style/nsStyleTransformMatrix.cpp	Mon Mar 04 20:33:13 2024 +0000
+++ b/layout/style/nsStyleTransformMatrix.cpp	Mon Mar 04 20:46:05 2024 +0000
@@ -101,7 +101,8 @@
       // FIXME: Bug 1849054. We may have to update
       // SVGGeometryFrame::GetBBoxContribution() to get tighter stroke bounds.
       nsRect strokeBox = nsLayoutUtils::ComputeSVGReferenceRect(
-          const_cast<nsIFrame*>(aFrame), StyleGeometryBox::StrokeBox);
+          const_cast<nsIFrame*>(aFrame), StyleGeometryBox::StrokeBox,
+          nsLayoutUtils::MayHaveNonScalingStrokeCyclicDependency::Yes);
       // The |nsIFrame::mRect| includes markers, so we have to compute the
       // offsets without markers.
       return nsRect{strokeBox.x - aFrame->GetPosition().x,